src/controller/tripController.go

```diff
@@ -379,99 +334,55 @@ func (tc *TripController) InviteUserToTrip(ctx context.Context, tripId *uuid.UUI
 		return nil, expenseerror.EXPENSE_FORBIDDEN
 	}
 
-	// Check if user to invite is already invited to trip
-	row = tc.DatabaseMgr.ExecuteQueryRow(checkUserTripQueryString, tripId, userId)
-	if err := row.Scan(&count); err != nil {
-		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteQueryRow(): %v", err)
-		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
-	}
-
-	if count != 0 {
-		return nil, expenseerror.EXPENSE_BAD_REQUEST
-	}
-
-	// Get trip details
-	tripDetails := &model.TripResponse{}
-	getTripDetailsQueryString := "SELECT id, location, start_date, end_date FROM trip WHERE id = $1"
-	row = tc.DatabaseMgr.ExecuteQueryRow(getTripDetailsQueryString, tripId)
-	if err := row.Scan(&tripDetails.TripID, &tripDetails.Location, &tripDetails.StartDate, &tripDetails.EndDate); err != nil {
-		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteQueryRow(): %v", err)
-		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
-	}
-
 	// Add user to trip
-	addUserToTripQueryString := "INSERT INTO user_trip_association (id_trip, id_user, is_accepted, presence_start_date, presence_end_date) VALUES ($1, $2, $3, $4, $5)"
-	if _, err := tc.DatabaseMgr.ExecuteStatement(addUserToTripQueryString, tripId, userId, false, tripDetails.StartDate, tripDetails.EndDate); err != nil {
+	addUserToTripQueryString := "INSERT INTO user_trip_association (id_trip, id_user, is_accepted) VALUES ($1, $2, $3)"
+	if _, err := tc.DatabaseMgr.ExecuteStatement(addUserToTripQueryString, tripId, userId, false); err != nil {
+		// if err is unique_violation, then user is already part of trip
+		if pqErr, ok := err.(*pq.Error); ok {
+			if pqErr.Code == "23505" {
+				return nil, expenseerror.EXPENSE_CONFLICT
+			}
+		}
 		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteStatement(): %v", err)
 		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
```

Comment:
Weiß noch nicht, ob ich das schöner finde, aber kann noch etwas vereinfacht werden, um zumindest eine Ebene an Komplexität wegzunehmen.
```suggestion
	if _, err := tc.DatabaseMgr.ExecuteStatement(addUserToTripQueryString, tripId, userId, false); err != nil {
		// if err is unique_violation, then user is already part of trip
		if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == "23505" {
			return nil, expenseerror.EXPENSE_CONFLICT
		}
		log.Printf("Error in tripController.InviteUserToTrip.DatabaseMgr.ExecuteStatement(): %v", err)
		return nil, expenseerror.EXPENSE_UPSTREAM_ERROR
```